### ISSUE TITLE
MD051: False-positive if the header itself is a link

### ISSUE THREAD

**Issue by sschuberth** [DavidAnson/markdownlint#945]
In our [README](https://github.com/oss-review-toolkit/ort/blob/main/README.md), code like

    #### [ORT configuration file](./model/src/main/resources/reference.yml)

triggers [MD051](https://github.com/DavidAnson/markdownlint/blob/v0.30.0/doc/md051.md) for a reference like

    [ORT configuration file](#ort-configuration-file)

I guess this happens because the linter does not recognize the link syntax in the heading.

PS: This seem like a regression in markdownlint v0.30.0 as markdownlint v0.29.0 did not complain.

**Comment by philrz** [comment]
It looks like we've bumped into this one as well. We've got a markdown file at https://github.com/brimdata/zed/blob/main/docs/tutorials/schools.md that's always checked clean in the past but started showing these MD051 failures once markdownlint v0.30.0 was in use.

In our case a reference looks like:

```
[Boolean `and`](#541-and)
```

with the corresponding header looking like:

```
#### 5.4.1 [and](../language/aggregates/and.md)
```

**Comment by DavidAnson** [comment]
Good timing, @philrz - I verified that your example is fixed by the change I just made for the original example by @sschuberth.
